src/selectors/modular-pipelines.js

```diff
@@ -0,0 +1,59 @@
+import { createSelector } from 'reselect';
+import { getPipelineModularPipelineIDs } from './pipeline';
+
+const getNodesModularPipelines = (state) => state.node.modularPipelines;
+const getModularPipelineIDs = (state) => state.modularPipeline.ids;
+const getModularPipelineName = (state) => state.modularPipeline.name;
+const getModularPipelineEnabled = (state) => state.modularPipeline.enabled;
+
+/**
+ * Retrieve the formatted list of modular pipeline filters
+ */
+export const getModularPipelineData = createSelector(
+  [getModularPipelineIDs, getModularPipelineName, getModularPipelineEnabled],
+  (modularPipelineIDs, modularPipelineName, modularPipelineEnabled) =>
+    modularPipelineIDs
+      .slice()
+      .sort()
+      .map((id) => ({
+        id,
+        name: modularPipelineName[id],
+        enabled: Boolean(modularPipelineEnabled[id]),
+      }))
+);
+
+/**
+ * Get the total and enabled number of modular pipelines
+ */
+export const getModularPipelineCount = createSelector(
+  [getPipelineModularPipelineIDs, getModularPipelineEnabled],
+  (modularPipelineIDs, modularPipelineEnabled) => ({
+    total: modularPipelineIDs.length,
+    enabled: modularPipelineIDs.filter((id) => modularPipelineEnabled[id])
+      .length,
+  })
+);
+
+/**
+ * returns an array of modular pipelines with the corresponding
+ * nodes for each modular pipeline
+ */
+export const getModularPipelineNodes = createSelector(
+  [getNodesModularPipelines, getModularPipelineIDs],
+  (allNodesModularPipelines, modularPipelines) => {
+    return modularPipelines.map((modularPipeline) => {
+      let nodes = [];
+      Object.keys(allNodesModularPipelines).forEach((key) => {
+        if (
+          allNodesModularPipelines[key] &&
+          allNodesModularPipelines[key].includes(modularPipeline)
+        )
+          nodes.push(key);
+      });
+      return {
+        modularPipeline,
+        nodes,
+      };
+    });
+  }
+);
```

Comment:
@studioswong Hey I merged this branch into the one I'm working on, and realised that this selector isn't actually being used anywhere? Anyway I've deleted it on my branch.